src/openforms/contrib/kvk/client.py

```diff
@@ -81,7 +79,21 @@ def get_search_results(self, query_params: SearchParams):
             )
             response.raise_for_status()
         except requests.RequestException as exc:
-            logger.exception("exception while making KVK request", exc_info=exc)
+            logger.exception("exception while making KVK zoeken request", exc_info=exc)
             raise exc
 
         return response.json()
+
+
+def get_kvk_client(
+    client: Literal["profile", "search"]
+) -> KVKProfileClient | KVKSearchClient:
+    service_mappings = {
+        "profile": KVKProfileClient,
+        "search": KVKSearchClient,
+    }
+    config = KVKConfig.get_solo()
+    assert isinstance(config, KVKConfig)
+    if not (service := getattr(config, f"{client}_service")):
+        raise NoServiceConfigured(f"No KVK {client} service configured!")
+    return build_client(service, client_factory=service_mappings[client])
```

Comment:
I think it would be easier to split this helper function in two ones. Without the use of overloads, it's not possible to know the return type from the `client` argument. So perhaps something like `get_kvk_profile_client` and `get_kvk_search_client`

Reply:
It was my first thought but then I realized that they were almost identical that's why I did that in one function.
In my opinion, what you suggest is more clear so I will fix that.